### PR TITLE
Improve `gluonts.dataset.split`.

### PR DESCRIPTION
**Pull request by jaheba**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/gluonts/dataset/split/splitter.py [hunk 1]
```diff
@@ -49,64 +49,95 @@
...
+        feat_dynamic_cat,
+        feat_dynamic_real,
+    ):
+        self.freq = freq
+        self.index = index
+        self.target = target
+        self.feat_static_cat = feat_static_cat
+        self.feat_static_real = feat_static_real
+        self.feat_dynamic_cat = feat_dynamic_cat
+        self.feat_dynamic_real = feat_dynamic_real
```
Comment (lostella) [inline src/gluonts/dataset/split/splitter.py:75]: dataclass?

Comment (jaheba) [reply src/gluonts/dataset/split/splitter.py:75]: requires 3.7

Comment (lostella) [reply src/gluonts/dataset/split/splitter.py:75]: pydantic then?

**Review by lostella — COMMENTED** [review]
Looks good to me, should this be labeled as breaking?

**Pull request closed without merging**